mongoexport/mongoexport.go

```diff
@@ -378,6 +408,27 @@ func (exp *MongoExport) exportInternal(out io.Writer) (int64, error) {
 // of documents successfully exported, and a non-nil error if something went wrong
 // during the export operation.
 func (exp *MongoExport) Export(out io.Writer) (int64, error) {
+	session, err := exp.SessionProvider.GetSession()
```

Comment:
the docstring for `exportInternal` claims it is used directly in testing which would bypass this new logic. We then may want to move this code into `exportInternal` so existing and new tests written that way will exercise it.

Reply:
Moved to exportInternal.